pkg/cloud/services/ec2/eips.go

```diff
@@ -98,7 +98,12 @@ func (s *Service) releaseAddresses() error {
 
 	for _, ip := range out.Addresses {
 		if ip.AssociationId != nil {
-			return errors.Errorf("failed to release elastic IP %q with allocation ID %q: Still associated with association ID %q", *ip.PublicIp, *ip.AllocationId, *ip.AssociationId)
+			_, err := s.scope.EC2.DisassociateAddress(&ec2.DisassociateAddressInput{
```

Comment:
Should this use `wait.WaitForWithRetryable`?

Reply:
Can do, or it will error and come back on next reconcile if it fails. Disassociation seemed to be immediate from my tests though.

Reply:
Can we move this block and if check in the retryable below before ReleaseAddress?

Reply:
I think that would assume they both return the same "retryable" errors, I'm not sure that is the case.

Reply:
Resolved this by moving to its own retryable function where it handles the errors specific to it.